--- modules/demux/ts.c.orig	2014-06-02 08:38:32.688618168 -0500
+++ modules/demux/ts.c	2014-06-02 15:41:41.888301082 -0500
@@ -278,6 +278,7 @@
     int             i_number;
     int             i_pid_pcr;
     int             i_pid_pmt;
+    int             i_number_alternate;
     /* IOD stuff (mpeg4) */
     iod_descriptor_t *iod;
 
@@ -1485,6 +1486,14 @@
             {
                 i_pmt_pid = p_sys->pmt[i]->i_pid;
                 p_prg = p_sys->pmt[i]->psi->prg[i_prg];
+                if( p_prg->i_number_alternate >= 0 )
+                {
+                    p_sys->i_current_program = p_prg->i_number_alternate;
+                    SetPrgFilter( p_demux, p_prg->i_number_alternate, b_selected );
+                    es_out_Control( p_demux->out, ES_OUT_SET_GROUP, p_prg->i_number_alternate );
+                    msg_Dbg( p_demux, "Setting filter for alternate program %d", p_prg->i_number_alternate);
+                    return;
+                }
                 break;
             }
         }
@@ -1556,6 +1565,7 @@
                 prg->i_number   = -1;
                 prg->i_pid_pcr  = -1;
                 prg->i_pid_pmt  = -1;
+                prg->i_number_alternate = -1;
                 prg->iod        = NULL;
                 prg->handle     = NULL;
 
@@ -4215,8 +4225,17 @@
 
         if( !old_pid && p_sys->pid[p_es->i_pid].b_valid )
         {
+            ts_pid_t *other_pid = &p_sys->pid[p_es->i_pid];
             msg_Warn( p_demux, "pmt error: pid=%d already defined",
                       p_es->i_pid );
+            if( other_pid->p_owner && other_pid->es &&
+                    ( p_es->i_type == 0x02 || p_es->i_type == 0x03 ||
+                      p_es->i_type == 0x04 || p_es->i_type == 0x10 )
+                    )
+            {
+                prg->i_number_alternate = other_pid->p_owner->prg[0]->i_number;
+                msg_Dbg( p_demux, "Storing alternate program %d for program %d with same pids", prg->i_number_alternate, prg->i_number);
+            }
             continue;
         }
 
