Transport Stream Demux - avoid traversing invalid PIDs
Index: vlc-2.0.8/modules/demux/ts.c
===================================================================
--- vlc-2.0.8.orig/modules/demux/ts.c	2013-06-17 17:07:53.000000000 -0500
+++ vlc-2.0.8/modules/demux/ts.c	2014-02-18 14:36:59.858068546 -0600
@@ -1478,7 +1478,8 @@
     for( int i = 0; i < p_sys->i_pmt; i++ )
     {
         ts_pid_t *pmt = p_sys->pmt[i];
-
+	if(pmt->b_valid != true) // REW: don't run down invalid programs
+            continue;
         for( int i_prg = 0; i_prg < pmt->psi->i_prg; i_prg++ )
         {
             if( pmt->psi->prg[i_prg]->i_number == i_prg_id )
@@ -2188,6 +2189,8 @@
         /* Search program and set the PCR */
         for( int i = 0; i < p_sys->i_pmt; i++ )
         {
+            if(p_sys->pmt[i]->b_valid != true)  // REW - don't run down invalid pmt
+                 continue;
             for( int i_prg = 0; i_prg < p_sys->pmt[i]->psi->i_prg; i_prg++ )
             {
                 if( pid->i_pid == p_sys->pmt[i]->psi->prg[i_prg]->i_pid_pcr )
@@ -4101,6 +4104,8 @@
     for( int i = 0; i < p_sys->i_pmt; i++ )
     {
         int i_prg;
+	if(p_sys->pmt[i]->b_valid != true)
+            continue;	//REW - Don't run down invalid PMT
         for( i_prg = 0; i_prg < p_sys->pmt[i]->psi->i_prg; i_prg++ )
         {
             const int i_pmt_number = p_sys->pmt[i]->psi->prg[i_prg]->i_number;
